tests/unit/workflows/test_workflows_tasks_submission.py

```diff
@@ -22,22 +22,380 @@
+
+
+@mock.patch('inspirehep.modules.workflows.tasks.submission.submit_rt_ticket')
+@mock.patch('inspirehep.modules.workflows.tasks.submission.render_template')
+@mock.patch('inspirehep.modules.workflows.tasks.submission.User')
+def test_create_ticket(mock_user_query, mock_render_template, mock_submit_rt_ticket):
+    template = 'literaturesuggest/tickets/curator_submitted.html'
+    ticket_id_key = 'ticket_id'
+    queue = 'HEP_add_user'
+    create_ticket_function = create_ticket(
+        template=template,
+        queue=queue,
+        context_factory=mock_context_function,
+        ticket_id_key=ticket_id_key
+    )
+
+    mock_user = MagicMock()
+    mock_user.email.return_value = ''
+    mock_user_query.query.get.return_value = mock_user
+    mock_render_template.return_value = ''
+
+    obj = MockObj(1, {}, {})
+    create_ticket_function(obj, {})
+
+    mock_render_template.assert_called_with(template)
+    mock_submit_rt_ticket.assert_called_with(obj, queue, None, '',
+                                             mock_user.email, ticket_id_key)
+
+
+@pytest.mark.parametrize('mock_template', [None, MagicMock()])
+@mock.patch('inspirehep.utils.tickets.get_instance')
+@mock.patch('inspirehep.modules.workflows.tasks.submission.User')
+@mock.patch('inspirehep.modules.workflows.tasks.submission.render_template')
+def test_reply_ticket_with_template(mock_render_template, mock_user_query, mock_get_instance, mock_template):
+    mock_rt = MagicMock()
+    mock_get_instance.return_value = mock_rt
+    mock_template = mock_template
+    mock_render_template.return_value = 'Reason\ndescription'
+    mock_user_query.query.get.return_value = MagicMock()
+
+    reply_ticket_function = reply_ticket(
+        template=mock_template,
+        context_factory=mock_context_function,
+        keep_new=True
+    )
+
+    obj = MockObj(1, {}, {'ticket_id': 1, 'reason': 'Reason\ndescription '})
+    reply_ticket_function(obj, {})
+
+    mock_rt.edit_ticket.assert_called_with(ticket_id=1, Status='new')
+    mock_rt.reply.assert_called_with(ticket_id=1, text='Reason\n description')
+
+
+@mock.patch('inspirehep.utils.tickets.get_instance')
+def test_close_ticket(mock_get_instance):
+    mock_rt = MagicMock()
+    mock_get_instance.return_value = mock_rt
+    close_ticket_function = close_ticket(ticket_id_key='ticket_id')
+
+    obj = MockObj(1, {}, {'ticket_id': 1})
+    close_ticket_function(obj, {})
+
+    mock_rt.edit_ticket.assert_called_with(ticket_id=1, Status='resolved')
+
+
+@pytest.mark.parametrize('extra_data_key', [None, 'something'])
+@mock.patch('inspirehep.modules.workflows.tasks.submission.in_production_mode')
+@mock.patch('inspirehep.utils.robotupload.make_robotupload_marcxml')
+@mock.patch('inspirehep.dojson.utils.legacy_export_as_marc')
+def test_send_robotupload(mock_legacy_export_as_marc,
+                          mock_make_robotupload_marcxml,
+                          mock_production_mode,
+                          extra_data_key):
```

Comment:
Tip: when mock names become this unwieldy you can just use the initials of every letter: `mock_legacy_export_as_marc` -> `l_e_a_m`.

Although take note that, when that happens, maybe you're mocking too many things...